src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -826,48 +834,103 @@ toJSONPBMessageInstD _ctxt parentIdent msgIdent messageParts = do
               , HsVar (unqual_ varNm)
               ]
 
-  -- E.g.
-  -- case f4_or_f9 of
-  --   Just (SomethingPickOneName f4)
-  --     -> HsJSONPB.pair "name" f4
-  --   Just (SomethingPickOneSomeid f9)
-  --     -> HsJSONPB.pair "someid" f9
-  --   Nothing
-  --     -> mempty
-  let oneofCaseE (OneofField{subfields}) =
-        HsCase disjunctName (altEs <> pure fallThruE)
+  -- Note: [Generating eta-reduced code]
```

Comment:
What is the benefit of generating eta-reduced code?

Reply:
I think explicitness increases clarity in most cases by reducing indirection. There are cases where pointfree code is more natural to read and it makes sense; however, I think a lot of pointfree code introduces too much indirection, reducing clarity. 

I think clarity is important because it lowers the "gumption barrier" for contributors familiar and not familiar with the codebase. Clarity is also important for users who may have to debug issues or figure out how to use the generated code.

The fact that we have complete control over what is generated only reinforces that we should do the most clear thing for our users and for contributors.